### DISCUSSION TITLE
Kafka-ui not loading on Kubernetes/Rancher

### DISCUSSION THREAD

**Discussion by cetinkesepara** [provectus/kafka-ui#1551] [Q&A] [answered]
Hi,
I can use Kafka-ui on Docker. But when I build it on Kubernetes with Rancher, the page doesn't load.

It is constantly loading as below.

![k1](https://user-images.githubusercontent.com/59224762/152535593-0cd78138-787b-419c-951f-52b96c0aae1c.png)

Console output

![2](https://user-images.githubusercontent.com/59224762/152535616-1b08cab1-8487-4747-8cd5-df6335c9458c.png)

What is the problem?

**Comment by Haarolean** [comment, marked as answer]
Hi, the issue seems definitely in your configuration. Considering that your app is somehow located at `xx:8080/proxy` and there is a bunch of 404s, I suggest you haven't set the `SERVER_SERVLET_CONTEXT_PATH` variable. 

Also, check out [this](https://github.com/provectus/kafka-ui/issues/1343) issue, this might help.

**Reply by cetinkesepara** [reply]
This didn't help me. I reach the page you see with the Cluster IP service. No Ingress has yet been defined. So it's normal to see it as `xx:8080/proxy`. Only when I go to this page, it cannot pull the data in it somehow.

I can also access `xx:8080/proxy/actuator, xx:8080/proxy/actuator/info`.

There is currently no Kafka Environment in the Pod settings. I just want to see a blank UI page. 

The image below is the Kafka-ui image I'm running with Docker. I can't get this result in Kubernetes.

![asd](https://user-images.githubusercontent.com/59224762/152561681-abced87f-7b80-4daa-a6c3-2c218022c794.png)

[the remark this answers is not in the document]
**Reply by cetinkesepara** [reply]
Anyone have any ideas? Otherwise, we will unfortunately stop using Kafka-ui.

[the remark this answers is not in the document]
**Reply by 5hin0bi** [reply]
Hey @cetinkesepara, could you please also provide browser F12 console log of the Network tab when opening kafka-ui in K8s?

 @Haarolean it loads the main UI page itself but cannot load specific page modules. It's not rly clear why it happens. I think Network tab would be useful.
 
 My assumption is that /proxy thingy somehow affects the working process.

[the remark this answers is not in the document]
**Reply by 5hin0bi** [reply]
@cetinkesepara kafka-ui pod log would be also appreciated. From the moment the pod is started and till the moment you have tried to access the main page.

[the remark this answers is not in the document]
**Reply by cetinkesepara** [reply]
@5hin0bi 
Network:
![1](https://user-images.githubusercontent.com/59224762/152944137-4095cf0e-33c7-4a51-9e15-c7dae7420b44.png)

clusters:
`kind: "Status", apiVersion: "v1", metadata: {}, status: "Failure",…}
apiVersion: "v1"
code: 404
details: {}
kind: "Status"
message: "the server could not find the requested resource"
metadata: {}
reason: "NotFound"
status: "Failure"`

Manifest:
`{type: "error", status: "401", message: "Unauthorized 401: must authenticate"}
message: "Unauthorized 401: must authenticate"
status: "401"
type: "error"`

Pod Log: There is currently no Kafka Environment in the Pod settings. I just want to see a blank UI page.
![2](https://user-images.githubusercontent.com/59224762/152944170-0cbf31c5-a694-4687-bcd0-0c3de3083d98.png)

[the remark this answers is not in the document]
**Reply by 5hin0bi** [reply]
@cetinkesepara thank you for the provided information.

Let me try to conclude on what we see here:

1. Basing on the Network tab information - URLs that were requested differ by size. The first is much longer than the second. This should not be like this. First URL contains port, second one does not.
The first request targets _8080/proxy_, the second targets the URL without _/proxy_ path. The reason is that the **application is most likely not ready to work with the /proxy path**. To solve it you may set _SERVER_SERVLET_CONTEXT_PATH_=/proxy as @Haarolean suggested before. This will tell the application to return currently not found  _manifest.json_ and _/api/clusters_ by the _/proxy_ path.

3. Since our helm chart provides the possibility to use ingress you can try to use it. This can resolve your problem as well.

I hope this helps. In case of any further questions please let us know.

[the remark this answers is not in the document]
**Reply by cetinkesepara** [reply]
@5hin0bi  Thanks but it didn't work. I did what you said. This time, I was able to reach the same page as follows.

`:8080/proxy/proxy/`

[the remark this answers is not in the document]
**Reply by 5hin0bi** [reply]
@cetinkesepara what do you see on the F12 Network tab now? Can you please attach a screenshot again?

Can you try to access Kafka-UI via Kafka-UI service port-forwarding? Will it work?
I've taken a look at the k8s configuration in the demo env we have and SVC for Kafka-UI is of the Cluster IP type and works fine. I don't have to use /proxy path.
Is it possible to see your SVC yaml file?

Please let me know.

[the remark this answers is not in the document]
**Reply by cetinkesepara** [reply]
@5hin0bi 
First of all, we use Rancher for Kubernetes. While creating the pod, I pull the image here "https://hub.docker.com/r/provectuslabs/kafka-ui" and opened port 8080. Then I set up a Cluster_ip service and connected it to the Pod.

I can't deal with this anymore. Thanks for your help.
